### PR TITLE
Latest: wrap Possiblies as Array (Sum Error t)

### PR DESCRIPTION
**Pull request by amosr**

### CODE REVIEW

**Comment by amosr** [comment]
cool

File: src/Icicle/Source/ToCore/Fold.hs [hunk 1]
```diff
@@ -236,81 +236,23 @@ convertFold q
...
+                     ( CE.pushBuf t'e
+                         CE.@~ CE.xVar n'acc
+                         CE.@~ res )
+           let zero  = CE.emptyBuf t'e
+                         CE.@~ CE.constI i
+
+           let x'    = CE.xLam n'buf t'buf
+                     ( CE.readBuf t'e CE.@~ CE.xVar n'buf )
+
+           return $ ConvertFoldResult kons zero x' t'buf t'arr
```
Comment (tranma) [inline src/Icicle/Source/ToCore/Fold.hs]: cool, this was pretty a bit much

**Comment by HuwCampbell** [comment]
Will this fix #157 ?

File: src/Icicle/Source/ToCore/Fold.hs (hunk 1 above)
Comment (amosr) [inline src/Icicle/Source/ToCore/Fold.hs]: this is true

Comment (amosr) [inline src/Icicle/Source/ToCore/Fold.hs]: yeah, it's also a shame that this stuff can't all be done in source to source right now

**Comment by amosr** [comment]
not sure..

**Comment by amosr** [comment]
I think I'll merge this

**Pull request merged**